### ISSUE TITLE
Adding a bunch of default icons in a directory for users to choose from

### ISSUE THREAD

**Issue by Urban-Hacker** [sparrowwallet/sparrow#1075]
Sparrow enables users to choose a custom icon for their wallet. When selecting the option it opens the file browser but this requires people to know where their operating system stores icons most of them are not suitable for a wallet.

Taking inspiration from various banking app we could provide a series of default icons in both dark and light mode, they could be placed in a directory and the file browser would open this repository by default.

Included bellow a screenshot to show how this looks:
![image](https://github.com/sparrowwallet/sparrow/assets/92275709/399788c2-30e8-4fcc-ad7d-9a8a95940251)

Icons have several advantages, usually people will recognize them quicker than text.

The current icon set I included comes from https://pictogrammers.com/library/mdi/ and contains 15 hand picked icons.

Car.png
Cold Storage.png
Energy.png
Entertainment.png
Family.png
Flower.png
Food.png
Gift.png
Health Care.png
Lightning.png
Ordinals.png
Phone.png
Shopping.png
Travel.png
Whirlpool.png

If someone has ideas of other icons we can include feel free to suggest. I think what is important is not which category we include or not but having a small series to choose from. It is also better if we do not put too many icons (I think 20 is a good upper limit) to not overwhelm with too many choices.

Regarding the dark theme, we can simply invert the icons. We could also propose a series of color scheme (eg: blue, red, green, dark, light) so you can use colors to differentiate too. I can provide all the icons set needed but as long as we do not agree on a set I will not spend time on it.

Icon set: [SparrowIcons.zip](https://github.com/sparrowwallet/sparrow/files/12402543/SparrowIcons.zip)

**Comment by craigraw** [comment]
Thanks for the suggestion. My apologies, I should have been clearer on requirements previously:

- I'd prefer icons from https://fontawesome.com/ to match those already present in the app
- Icons must be SVG, 15px x 15px
- Light theme icons must be dark gray, not black
- Dark theme icons must be light gray, not white

For example, see 
Light theme: https://github.com/sparrowwallet/sparrow/blob/c57a4450467aef03e248504db0d935283a46b136/src/main/resources/image/bitbox02-icon.svg
Dark theme: https://github.com/sparrowwallet/sparrow/blob/c57a4450467aef03e248504db0d935283a46b136/src/main/resources/image/bitbox02-icon-invert.svg

There are other examples in that folder too.

I think a small set of icons (6-7) is probably enough for now. Likely these would be in a menu drop-down.

**Comment by Urban-Hacker** [comment]
Hi, sorry for the very long delay before answering but I followed your guidelines and I think we have got something to start with.

So the current categories are:
Cold Storage
Food
Mining
Ordinals
Entertainment
Healthcare

![image](https://github.com/sparrowwallet/sparrow/assets/92275709/e624e910-e274-4600-827e-8d54a8f54a37)


Of course if requires we can always add more. For now I only did the dark grey version for the light theme. If we agree on it I will export them again for the dark mode.

Another thing I noticed while I was playing with those icons. The size of 15px is really small, I think it would be beneficial if we increase it to 18 to have them slightly bigger.

Usually people are better at recognizing icons than text which is an advantage :)

[SparrowIcons.zip](https://github.com/sparrowwallet/sparrow/files/12838847/SparrowIcons.zip)

**Comment by craigraw** [comment]
Thanks for looking at this again. Re the size, I would be using them directly as vector TTF font characters, so they appear sharp regardless of size. The cold storage icon is perhaps the best, but conflicts with the existing frozen UTXO icon. The mining category seems useful, but I didn't immediately associate the icon with the category. I'll need to consider the other categories.

**Comment by Urban-Hacker** [comment]
Ah true did not think about the freezing UTXO. I could use the image of a safe or something similar but the resolution being rather small it was challenging to make them visually distinctive.

One way to maybe make this less "challenging" is to do it similar like my router is doing to label IoT devices. They just have a bunch of icons like flower, server, carrot, fan, computer, printer, etc and open the folder where they are. IF we add categories then we somehow need to define them but if its just a series of icon from a folder it is less critical.

**Comment by craigraw** [comment]
> bunch of icons like flower, server, carrot, fan, computer, printer, etc

Yes, but none of those icons seem particularly compelling. I'm not closing this though, I plan to take another look at the icon set.
